File: models/Gpt-3.5-turbo/py/code/introductory_problems/Gray_Code.py

def generate_gray_code(n):
    # Base case for n = 1
    if n == 1:
        return ['0', '1']
    
    # Recursive call to generate Gray code for n-1
    prev_gray_code = generate_gray_code(n-1)
    
    # Append '0' to the front of all strings in the previous Gray code
    gray_code = ['0' + code for code in prev_gray_code]
    
    # Reverse the previous Gray code and append '1' to the front of all strings
    gray_code += ['1' + code for code in reversed(prev_gray_code)]
    
    return gray_code

# Read the input
n = int(input())

# Generate the Gray code for n
gray_code = generate_gray_code(n)

# Print the Gray code
for code in gray_code:
    print(code)

File: models/Gpt-3.5-turbo/cpp/code/problems/introductory_problems/Gray_Code.cpp

#include <iostream>
#include <vector>
using namespace std;

void generateGrayCode(int n) {
    // Base case: Gray code for n = 1
    if (n == 1) {
        cout << "0\n1\n";
        return;
    }

    // Recursive case: Generate Gray code for n-1
    generateGrayCode(n - 1);

    // Append the reverse of Gray code for n-1
    vector<string> grayCode = {"0", "1"};
    for (int i = grayCode.size() - 1; i >= 0; i--) {
        cout << grayCode[i] << "\n";
        grayCode.push_back("1" + grayCode[i]);
        grayCode[i] = "0" + grayCode[i];
    }
}

int main() {
    int n;
    cin >> n;

    generateGrayCode(n);

    return 0;
}
